main.go

```diff
@@ -155,14 +155,18 @@ func runServer(
 	}
 
 	// Set up Opentracing and Jaeger tracer
+	var localAgentHostPort string
+	if (os.Getenv("JAEGER_AGENT_HOST") != "") && (os.Getenv("JAEGER_AGENT_PORT") != "") {
+		localAgentHostPort = fmt.Sprintf("%v:%v", os.Getenv("JAEGER_AGENT_HOST"), os.Getenv("JAEGER_AGENT_PORT"))
+	}
 	cfg := &jaegerconfig.Configuration{
 		Sampler: &jaegerconfig.SamplerConfig{
 			Type:  "const",
 			Param: 1,
 		},
 		Reporter: &jaegerconfig.ReporterConfig{
 			LogSpans:           true,
-			LocalAgentHostPort: fmt.Sprintf("%v:%v", os.Getenv("JAEGER_AGENT_HOST"), os.Getenv("JAEGER_AGENT_PORT")),
```

Comment:
Does this mean the cause of `tracer=nil` is `LocalAgentHostPort == ":"`?

Reply:
Yes, invalid for `LocalAgentHostPort`.